hubblestack/extmods/fdg/curl.py

```diff
@@ -111,36 +112,58 @@ def request(url,
         kwargs['auth'] = (username, password)
     if verify is not None:
         kwargs['verify'] = verify
+    if headers is not None:
+        kwargs['headers'] = headers
     kwargs['timeout'] = int(timeout)
     if function not in ('GET', 'PUT', 'POST'):
-        log.error('Invalid request type {0}'.format(function))
+        LOG.error('Invalid request type %s', function)
         return False, {}
 
     # Make the request
+    status, response = __make_request(function, url, **kwargs)
+    if not status:
+        return status, response
+
+    # Pull out the pieces we want
+    ret = _parse_response(response, decode_json)
+
+    # Status in the return is based on http status
+    try:
+        response.raise_for_status()
+        return True, ret
+    except requests.exceptions.HTTPError:
+        return False, ret
+
+
+def __make_request(function, url, **kwargs):
```

Comment:
Can we make this with a single leading underscore? I prefer the convention of single underscores to the name mangling of double underscore, unless we actually need the mangling. [more info](https://dbader.org/blog/meaning-of-underscores-in-python)